From 5aaa032129368cd0dd45c49263daf101378f1809 Mon Sep 17 00:00:00 2001
From: Dave Kover <dkover@cyngn.com>
Date: Tue, 3 Mar 2015 15:53:44 -0800
Subject: [PATCH 02/26] SystemUI: Expose values for themes.

Allow theming of volume panel separately from status. Allow
status bar icons to be tinted. Expose battery meter base color.
Expose status bar header elevation value. Add themeable ripple
color for navigation button press.

Change-Id: Ie3dbde82692a19927ba6c1da179e5d2f03391f6f
---
 packages/SystemUI/res/layout/status_bar_expanded_header.xml          | 2 +-
 packages/SystemUI/res/layout/volume_dialog.xml                       | 2 +-
 packages/SystemUI/res/layout/zen_mode_panel.xml                      | 2 +-
 packages/SystemUI/res/values/arrays.xml                              | 2 +-
 .../SystemUI/src/com/android/systemui/statusbar/BaseStatusBar.java   | 3 ++-
 .../src/com/android/systemui/statusbar/policy/KeyButtonRipple.java   | 5 ++++-
 6 files changed, 10 insertions(+), 6 deletions(-)
 mode change 100644 => 100755 packages/SystemUI/src/com/android/systemui/statusbar/BaseStatusBar.java

diff --git a/packages/SystemUI/res/layout/status_bar_expanded_header.xml b/packages/SystemUI/res/layout/status_bar_expanded_header.xml
index a3887cb..9d38442 100644
--- a/packages/SystemUI/res/layout/status_bar_expanded_header.xml
+++ b/packages/SystemUI/res/layout/status_bar_expanded_header.xml
@@ -26,7 +26,7 @@
     android:paddingStart="@dimen/notification_side_padding"
     android:paddingEnd="@dimen/notification_side_padding"
     android:baselineAligned="false"
-    android:elevation="4dp"
+    android:elevation="@dimen/status_bar_expanded_header_elevation"
     android:background="@drawable/notification_header_bg"
     android:clickable="true"
     android:focusable="true"
diff --git a/packages/SystemUI/res/layout/volume_dialog.xml b/packages/SystemUI/res/layout/volume_dialog.xml
index 3765fe8..c313d27 100644
--- a/packages/SystemUI/res/layout/volume_dialog.xml
+++ b/packages/SystemUI/res/layout/volume_dialog.xml
@@ -19,7 +19,7 @@
     android:layout_height="wrap_content"
     android:layout_marginLeft="@dimen/notification_side_padding"
     android:layout_marginRight="@dimen/notification_side_padding"
-    android:background="@drawable/qs_background_primary"
+    android:background="@drawable/volume_dialog_background"
     android:translationZ="@dimen/volume_panel_z"
     android:layout_marginBottom="@dimen/volume_panel_z">
 
diff --git a/packages/SystemUI/res/layout/zen_mode_panel.xml b/packages/SystemUI/res/layout/zen_mode_panel.xml
index 3d17820..5026c9c 100644
--- a/packages/SystemUI/res/layout/zen_mode_panel.xml
+++ b/packages/SystemUI/res/layout/zen_mode_panel.xml
@@ -28,7 +28,7 @@
         android:layout_height="wrap_content"
         android:minHeight="8dp"
         android:elevation="4dp"
-        android:background="@drawable/qs_background_secondary" >
+        android:background="@drawable/zen_mode_panel_background" >
 
         <com.android.systemui.volume.SegmentedButtons
             android:id="@+id/zen_buttons"
diff --git a/packages/SystemUI/res/values/arrays.xml b/packages/SystemUI/res/values/arrays.xml
index 6102aa6..79a4649 100644
--- a/packages/SystemUI/res/values/arrays.xml
+++ b/packages/SystemUI/res/values/arrays.xml
@@ -26,7 +26,7 @@
     </array>
     <array name="batterymeter_color_values">
         <item>@*android:color/battery_saver_mode_color</item>
-        <item>#FFFFFFFF</item>
+        <item>@color/batterymeter_base_color</item>
     </array>
     <array name="batterymeter_bolt_points">
         <item>73</item> <item>0</item>
diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/BaseStatusBar.java b/packages/SystemUI/src/com/android/systemui/statusbar/BaseStatusBar.java
old mode 100644
new mode 100755
index 8a03a2b..14dff4e
--- a/packages/SystemUI/src/com/android/systemui/statusbar/BaseStatusBar.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/BaseStatusBar.java
@@ -778,7 +778,8 @@ public abstract class BaseStatusBar extends SystemUI implements
 
         if (entry.icon != null) {
             if (entry.targetSdk >= Build.VERSION_CODES.LOLLIPOP) {
-                entry.icon.setColorFilter(mContext.getResources().getColor(android.R.color.white));
+                entry.icon.setColorFilter(mContext.getResources().getColor(
+                    R.color.notification_icon_color));
             } else {
                 entry.icon.setColorFilter(null);
             }
diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/policy/KeyButtonRipple.java b/packages/SystemUI/src/com/android/systemui/statusbar/policy/KeyButtonRipple.java
index 7c12811..863a907 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/policy/KeyButtonRipple.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/policy/KeyButtonRipple.java
@@ -66,16 +66,19 @@ public class KeyButtonRipple extends Drawable {
     private final HashSet<Animator> mRunningAnimations = new HashSet<>();
     private final ArrayList<Animator> mTmpArray = new ArrayList<>();
 
+    private int mRippleColor;
+
     public KeyButtonRipple(Context ctx, View targetView) {
         mMaxWidth =  ctx.getResources().getDimensionPixelSize(R.dimen.key_button_ripple_max_width);
         mTargetView = targetView;
+        mRippleColor = ctx.getResources().getColor(R.color.navbutton_ripple_color);
     }
 
     private Paint getRipplePaint() {
         if (mRipplePaint == null) {
             mRipplePaint = new Paint();
             mRipplePaint.setAntiAlias(true);
-            mRipplePaint.setColor(0xffffffff);
+            mRipplePaint.setColor(mRippleColor);
         }
         return mRipplePaint;
     }
-- 
1.9.1

